### PR TITLE
Adds bugsplat-js

### LINKED ISSUE #12: Acceptance tests can only be run against Octomore
Octomore is a BugSplat dev server that is only on part-time. We need to figure out what the CORS issue is on the production server so that we can Fred as the testDatabase in the acceptance tests.

### LINKED ISSUE #27: No example of how to create/attach a log file
There is an example in my-angular-4-crasher regarding how to create and attach an image, however we should make a convenient factory method that can create/attach a log file for the consumer.

### LINKED ISSUE #39: Add additional file can cause crashes to fail to post
If you add an additional file every time handle error is called you will eventually cause the post to be too large and fail

### LINKED ISSUE #52: Safari and Firefox Errors should include Error: message
When we grab the call stack here the error message is getting lost in both Safari and Firefox. We should generate the call stack in a way that's compatible with Safari and Firefox.

### LINKED ISSUE #53: Can we use bugsplat-js for posting crash reports?
It would be nice if we could leverage bugsplat-js for the bulk of sending the crash reports and only use this library as a wrapper that adds angular-specific functionality. By doing this we wouldn't have to fix https://github.com/BugSplat-Git/bugsplat-js/issues/34 in two places.

### PR DESCRIPTION
**Pull request by bobbyg603**
Fixes #53 Fixes #52 Fixes #39 Resolves #27 Resolves #12

### CODE REVIEW

**Comment by bobbyg603** [comment]
Blocked by https://github.com/BugSplat-Git/bugsplat-js/pull/45 and https://github.com/BugSplat-Git/bugsplat-js/pull/43 and a subsequent release/update

File: package.json [hunk 1]
```diff
@@ -32,10 +33,12 @@
     "@angular/cli": "^12.0.5",
     "@angular/compiler-cli": "^12.0.5",
     "@angular/language-service": "^12.0.5",
+    "@bugsplat/js-api-client": "^0.0.7",
```
Comment (bobbyg603) [inline package.json:36]: not needed anymore

File: package.json [hunk 2]
```diff
@@ -32,10 +33,12 @@
...
     "@types/jasmine": "~3.3.8",
     "@types/jasminewd2": "~2.0.6",
     "@types/node": "^12.11.1",
     "codelyzer": "^5.1.2",
+    "dotenv": "^10.0.0",
```
Comment (bobbyg603) [inline package.json:41]: not needed anymore

File: projects/bugsplat-ng/src/lib/bugsplat.ts [hunk 3]
```diff
@@ -1,92 +1,97 @@
...
 
 @Injectable()
 export class BugSplat {
-  public appKey: string = "";
-  public user: string = "";
-  public email: string = "";
-  public description: string = "";
 
-  public bugSplatPostEventSubject = new Subject<BugSplatPostEvent>();
+  // TODO BG replace with calls to bugsplat.setDefault
```
Comment (bobbyg603) [inline projects/bugsplat-ng/src/lib/bugsplat.ts:11]: Address TODO

Comment (bobbyg603) [reply projects/bugsplat-ng/src/lib/bugsplat.ts:11]: #63

File: projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts [hunk 4]
```diff
@@ -1,35 +1,13 @@
-import { HttpClient } from "@angular/common/http";
-import { ErrorHandler, Inject, Injectable, Optional } from '@angular/core';
+import { ErrorHandler, Injectable } from '@angular/core';
 import { BugSplat } from "./bugsplat";
-import { BugSplatConfiguration } from "./bugsplat-config";
-import { BugSplatLogger } from './bugsplat-logger';
 
 @Injectable()
 export class BugSplatErrorHandler implements ErrorHandler {
```
Comment (bobbyg603) [inline projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts:5]: no longer used, remove

File: projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts [hunk 5]
```diff
@@ -1,35 +1,13 @@
...
-  public bugsplat: BugSplat;
-
   static readonly ERROR_CANNOT_BE_NULL = "BugSplatErrorHandler Error: handleError was passed a null error!";
```
Comment (bobbyg603) [inline projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts:6]: no longer used, remove

File: projects/bugsplat-ng/package.json [hunk 6]
```diff
@@ -12,6 +12,7 @@
   },
   "peerDependencies": {
     "@angular/common": ">=5.0.0",
-    "@angular/core": ">=5.0.0"
+    "@angular/core": ">=5.0.0",
+    "bugsplat": "^7.0.2"
```
Comment (bobbyg603) [inline projects/bugsplat-ng/package.json:16]: we need to test this in a real app to see if peerDeps is okay or if it needs to be a real dependency

File: projects/bugsplat-ng/src/test/bugsplat.spec.ts [hunk 7]
```diff
@@ -1,91 +1,65 @@
...
-        bugsplat.addAdditionalFile(file);
-        expect(spy).toHaveBeenCalledWith(expectedMessage);
+        
+        const event = await promise;
+        expect(event.type).toEqual(BugSplatPostEventType.Error);
+        expect(event.responseData.success).toEqual(false);
+        expect(event.responseData.message).toContain('Bad Request');
     });
+
+    // TODO BG
```
Comment (bobbyg603) [inline projects/bugsplat-ng/src/test/bugsplat.spec.ts:53]: Re-enabled

File: projects/bugsplat-ng/package.json (hunk 6 above)
Comment (bobbyg603) [reply projects/bugsplat-ng/package.json]: 2cd1dca

File: projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts (hunk 5 above)
Comment (bobbyg603) [reply projects/bugsplat-ng/src/lib/bugsplat-error-handler.ts]: c3006f3

File: projects/bugsplat-ng/src/test/bugsplat.spec.ts (hunk 7 above)
Comment (bobbyg603) [reply projects/bugsplat-ng/src/test/bugsplat.spec.ts]: 22f3a01

File: package.json (hunk 2 above)
Comment (bobbyg603) [reply package.json]: 541c765

File: projects/bugsplat-ng/ng-package.json [hunk 8]
```diff
@@ -1,5 +1,6 @@
 {
   "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
+  "allowedNonPeerDependencies": ["bugsplat"],
```
Comment (bobbyg603) [inline projects/bugsplat-ng/ng-package.json:3]: Remove this

Comment (bobbyg603) [reply projects/bugsplat-ng/ng-package.json]: bdff13c

**Review by daveplunkett — APPROVED** [review]

**Pull request merged**